cumulusci/tasks/bulkdata/extract.py

```diff
@@ -91,90 +89,81 @@ def _init_mapping(self):
 
     def _run_query(self, soql, mapping):
-        self.logger.info(f"Creating bulk job for: {mapping['sf_object']}")
-        job = self.bulk.create_query_job(mapping["sf_object"], contentType="CSV")
-        self.logger.info(f"Job id: {job}")
-        self.logger.info(f"Submitting query: {soql}")
-        batch = self.bulk.query(job, soql)
-        self.logger.info(f"Batch id: {batch}")
-        self.bulk.wait_for_batch(job, batch)
-        self.logger.info(f"Batch {batch} finished")
-        self.bulk.close_job(job)
-        self.logger.info(f"Job {job} closed")
+        step = BulkApiQueryStep(mapping["sf_object"], {}, self, soql)
 
+        step.query()
+
+        if step.status is Status.SUCCESS:
+            self._import_results(mapping, step)
+        else:
+            raise BulkDataException("Bulk query failed")
+
+    def _import_results(self, mapping, step):
         conn = self.session.connection()
-        for result_file in self._get_results(batch, job):
-            self._import_results(mapping, result_file, conn)
-
-    def _get_results(self, batch_id, job_id):
-        result_ids = self.bulk.get_query_batch_result_ids(batch_id, job_id=job_id)
-        for result_id in result_ids:
-            self.logger.info(f"Result id: {result_id}")
-            uri = (
-                f"{self.bulk.endpoint}/job/{job_id}/batch/{batch_id}/result/{result_id}"
-            )
-            with download_file(uri, self.bulk) as f:
-                self.logger.info(f"Result {result_id} downloaded")
-                yield f
 
-    def _import_results(self, mapping, result_file, conn):
         # Map SF field names to local db column names
-        sf_header = [
-            name.strip('"')
-            for name in result_file.readline().strip().decode("utf-8").split(",")
-        ]
+        fields = self._fields_for_mapping(mapping)
         columns = []
         lookup_keys = []
-        for sf in sf_header:
-            if sf == "Records not found for this query":
-                return
-            if sf:
-                column = mapping.get("fields", {}).get(sf)
-                if not column:
-                    lookup = mapping.get("lookups", {}).get(sf, {})
-                    if lookup:
-                        lookup_keys.append(sf)
-                        column = get_lookup_key_field(lookup, sf)
-                if column:
-                    columns.append(column)
+        for sf in fields:
+            column = mapping.get("fields", {}).get(sf)
+            if not column:
+                lookup = mapping.get("lookups", {}).get(sf, {})
+                if lookup:
+                    lookup_keys.append(sf)
+                    column = get_lookup_key_field(lookup, sf)
+            if column:
+                columns.append(column)
+
         if not columns:
             return
         record_type = mapping.get("record_type")
         if record_type:
             columns.append("record_type")
 
-        processor = log_progress(
```

Comment:
Looks like we've lost the logging of progress as records are extracted.

Reply:
Restoring it, although I think when we add REST API support progress logging needs to be handled differently.